contracts/Vault.sol

```diff
@@ -64,9 +76,21 @@ contract Vault is ERC4626Metadata, Ownable {
     // If receiver is omitted, use the sender
     function deposit(uint256 assets, address receiver) public override returns (uint256) {
         if (receiver == address(0)) receiver = _msgSender();
+        // Set lock time
+        lockedTime[receiver] = block.timestamp + _lockDuration;
         return super.deposit(assets, receiver);
     }
 
+    // If receiver is omitted, use the sender
+    function withdraw(uint256 assets, address receiver, address owner) public override returns (uint256) {
+        if (receiver == address(0)) receiver = _msgSender();
```

Comment:
Does this impose any checks on the caller? Seems like anyone can just call whatever